dash/include/dash/GlobAsyncRef.h

```diff
@@ -320,33 +321,56 @@ std::ostream & operator<<(
 template<typename T>
 class Future<dash::GlobRef<T>> {
 public:
-  typedef dash::GlobRef<T> reference_t;
-  typedef T                value_t;
+  typedef dash::GlobRef<T>         reference_t;
+  typedef T                        value_t;
+  typedef Future<dash::GlobRef<T>> self_t;
 
 protected:
 
-  value_t       _value;
-  dart_handle_t _handle;
-  bool          _completed = false;
+  std::vector<value_t> _valptr    = nullptr;
+  dart_handle_t        _handle;
+  bool                 _completed = false;
 
 public:
 
   /**
    * Create a Future from a \ref GlobRef instance.
    */
-  Future(dash::GlobRef<T>& ref) {
-    dart_storage_t ds = dart_storage<T>(1);
-    dart_get_handle(&_value, ref.dart_gptr(), ds.nelem, ds.dtype, &_handle);
+  Future(dash::GlobRef<T>& ref, size_t count = 1)
+  : _valptr(count) {
+    dart_storage_t ds = dart_storage<T>(count);
+    dart_get_handle(
+      _valptr.data(),
+      ref.dart_gptr(),
+      ds.nelem, ds.dtype,
+      &_handle);
   }
 
   /**
    * Create a Future from a \ref GlobAsyncRef instance.
    */
-  Future(dash::GlobAsyncRef<T>& aref) {
-    dart_storage_t ds = dart_storage<T>(1);
-    dart_get_handle(&_value, aref.dart_gptr(), ds.nelem, ds.dtype, &_handle);
+  Future(dash::GlobAsyncRef<T>& aref, size_t count = 1)
```

Comment:
My reasoning was that I have to dynamically allocate the value buffer anyway (to make the Future movable) so I thought it would be a neat addition. But I agree that it does not really fit into the DASH concept.